### PR TITLE
raidboss: add Cenote Ja Ja Gural treasure dungeon

### PR DESCRIPTION
**Pull request by xiashtra**

### CODE REVIEW

**Comment by jacob-keller** [comment]
i'll be running some maps tomorrow and can give this a test run then to check for issues

**Review by jacob-keller — APPROVED** [review]
This has worked great for me the last few maps

**Review by jacob-keller — APPROVED** [review]

**Review by jacob-keller — APPROVED** [review]
I think this is ready to merge (tho I'm not sure my approval counts for the check)

**Comment by xiashtra** [comment]
> I think this is ready to merge (tho I'm not sure my approval counts for the check)

Unfortunately no, it'll need one of the other maintainers to approve first.

**Comment by valarnin** [comment]
I'm not familiar with the content, I've basically only done savage since it released and otherwise have been too busy to engage with other content, so I haven't been following this PR. I'll give it a once-over from a code standpoint tomorrow if another maintainer hasn't gotten to it by then.

File: ui/raidboss/data/07-dt/map/cenote_ja_ja_gural.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    {
+      id: 'Cenote Ja Ja Alpaca of Fortune Spawn',
+      // 13240 = Alpaca of Fortune
+      type: 'AddedCombatant',
+      netRegex: { npcNameId: '13240' },
+      suppressSeconds: 1,
+      infoText: (_data, matches, output) => output.spawn!({ name: matches.name }),
+      outputStrings: {
+        spawn: cenoteJaJaOutputStrings.spawn,
+      },
```
Comment (wexxlee) [inline ui/raidboss/data/07-dt/map/cenote_ja_ja_gural.ts:35]: I think you could just do this:

```suggestion
      outputStrings: cenoteJaJaOutputStrings,
```

File: ui/raidboss/data/07-dt/map/cenote_ja_ja_gural.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    {
+      id: 'Cenote Ja Ja Uolon of Fortune Spawn',
+      // 13241 = Uolon of Fortune
+      type: 'AddedCombatant',
+      netRegex: { npcNameId: '13241' },
+      suppressSeconds: 1,
+      infoText: (_data, matches, output) => output.spawn!({ name: matches.name }),
+      outputStrings: {
+        spawn: cenoteJaJaOutputStrings.spawn,
+      },
```
Comment (wexxlee) [inline ui/raidboss/data/07-dt/map/cenote_ja_ja_gural.ts:46]:
```suggestion
      outputStrings: cenoteJaJaOutputStrings,
```

File: ui/raidboss/data/07-dt/map/cenote_ja_ja_gural.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    },
+    {
+      id: 'Cenote Ja Ja Tuligora Troupes Spawn',
+      // 13242 = Turali Onion
+      // 13243 = Turali Eggplant
+      // 13244 = Turali Garlic
+      // 13245 = Turali Tomato
+      // 13246 = Tuligora Queen
+      type: 'AddedCombatant',
+      netRegex: { npcNameId: '1324[2-6]', capture: false },
```
Comment (wexxlee) [inline ui/raidboss/data/07-dt/map/cenote_ja_ja_gural.ts:56]: I don't remember the rationale (maybe @JLGarber does?), but I think we've been trying not to use regex character classes as values, and instead doing arrays, e.g.:
`const tuligoraTroupesIds = [13242, 13243, 13244, 13245, 13246];`
and then
`netRegex: { npcNameId: tuligoraTroupesIds, capture: false }`

**Review by wexxlee — APPROVED** [review]
Sorry for letting this linger.  I've been awol for a couple weeks with work travel, then getting super sick.

Haven't run this in game, but it sounds like @jacob-keller has, so this is just a style review.  LGTM - a few stylistic comments, feel free to take or leave.

**Pull request merged**